homeassistant/components/reolink/__init__.py

```diff
@@ -131,17 +131,15 @@ async def async_check_firmware_update() -> None:
         update_method=async_check_firmware_update,
         update_interval=FIRMWARE_UPDATE_INTERVAL,
     )
+
+    # If camera WAN blocked, firmware check fails and takes long, do not prevent setup
+    config_entry.async_create_task(hass, firmware_coordinator.async_refresh())
     # Fetch initial data so we have data when entities subscribe
-    results = await asyncio.gather(
-        device_coordinator.async_config_entry_first_refresh(),
-        firmware_coordinator.async_config_entry_first_refresh(),
-        return_exceptions=True,
-    )
-    # If camera WAN blocked, firmware check fails, do not prevent setup
-    # so don't check firmware_coordinator exceptions
-    if isinstance(results[0], BaseException):
+    try:
+        await device_coordinator.async_config_entry_first_refresh()
+    except BaseException:
```

Comment:
Probably unlikely but I guess this could raise CancelledError

Reply:
Yes the BaseException instead of Exception is to catch CancelledError just in case.
I am re-raising anyway, just need to make sure I always unregister the webhook (host.stop()), otherwise you will not be able to set-up again/recover. (you would then need a full HomeAssistant restart to recover).